docs/cluster-deployment-guide.md

```diff
@@ -363,7 +363,15 @@ If you are using a version that is older than 1.0.0, reach out to us at [communi
 First check to see if an upgrade to the database schema is required for the version you wish to upgrade to.
 If a database schema upgrade is required, it will be called out directly in the [release notes](https://github.com/temporalio/temporal/releases).
 Some releases require changes to the schema, and some do not.
-We ensure that any consecutive versions are compatible in terms of database schema upgrades, features, and system behavior, however there is no guarantee that there is compatibility between _any_ 2 non-consecutive versions.
+We ensure that any consecutive versions are compatible in terms of database schema upgrades, features, and system behavior, however there is no guarantee that there is compatibility between _any_ two non-consecutive versions.
+
+When upgrading your Temporal Server version, ensure that you upgrade sequentially. For example, when upgrading from v1.n.x, always upgrade to v1.n+1.x (or the next available version) and so on until you get to the required version.
+
+The Temporal Server upgrade updates or rewrites the old version data with the format introduced in the newer version. Since Temporal Server guarantees code backward compatibility between two consecutive minor versions, and because older versions of the code are eventually removed from the code base, skipping versions when upgrading may cause older formats to become unrecognizable. If the old format of the data can't be read to be rewritten to the new format, the upgrades will fail.
```

Comment:
suggest clearly referring to "persisted data" and less about code
```suggestion
The Temporal Server may update or rewrite persisted data with a format introduced in a newer version. Since Temporal Server guarantees backward compatibility between two consecutive minor versions, and because code to handle older versions of persisted data is eventually removed, skipping versions when upgrading may cause persisted data in an older format to become unrecognizable. In that case the server may fail to run or behave incorrectly.
```